File: Habitica/src/main/java/com/habitrpg/android/habitica/data/local/TutorialLocalRepository.java
package com.habitrpg.android.habitica.data.local;

import com.habitrpg.android.habitica.models.TutorialStep;

import java.util.List;

import io.realm.RealmResults;
import rx.Observable;

public interface TutorialLocalRepository extends BaseLocalRepository {

    Observable<TutorialStep> getTutorialStep(String key);
    Observable<RealmResults<TutorialStep>> getTutorialSteps(List<String> keys);
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/data/local/TutorialLocalRepository.kt
package com.habitrpg.android.habitica.data.local

import com.habitrpg.android.habitica.models.TutorialStep
import kotlinx.coroutines.flow.Flow

interface TutorialLocalRepository : BaseLocalRepository {

    fun getTutorialStep(key: String): Flow<TutorialStep>
    fun getTutorialSteps(keys: List<String>): Flow<List<TutorialStep>>
}
